runtime/shared_heap.c

```diff
@@ -286,37 +299,22 @@ static pool* pool_global_adopt(struct caml_heap_state* local, sizeclass sz)
       }
       #endif
 
-      calc_pool_stats(r, sz, &tmp_stats);
-      caml_accum_heap_stats(&local->stats, &tmp_stats);
-      caml_remove_heap_stats(&pool_freelist.stats, &tmp_stats);
-
-      if (local->stats.pool_words > local->stats.pool_max_words)
```

Comment:
This bit seems to have been left out of the refactor in `pool_global_adopt_stats` ?

Reply:
Actually it is now done in `caml_accum_heap_stats`, line 28, my bad.